cmd/root.go

```diff
@@ -151,35 +125,33 @@ func Run(c *cobra.Command, names []string) {
 		return
 	}
 
-	if err := actions.CheckForMultipleWatchtowerInstances(client, cleanup, scope); err != nil {
+	if err := actions.CheckForMultipleWatchtowerInstances(client, c.Cleanup, c.Scope); err != nil {
 		log.Fatal(err)
 	}
 
-	if httpAPI {
-		apiToken, _ := c.PersistentFlags().GetString("http-api-token")
-
-		if err := api.SetupHTTPUpdates(apiToken, func() { runUpdatesWithNotifications(filter) }); err != nil {
+	if c.HTTPAPI {
+		if err := api.SetupHTTPUpdates(c.HTTPAPIToken, func() { runUpdatesWithNotifications(filter) }); err != nil {
 			log.Fatal(err)
 			os.Exit(1)
 		}
 
 		api.WaitForHTTPUpdates()
 	}
 
-	if err := runUpgradesOnSchedule(c, filter); err != nil {
+	if err := runUpgradesOnSchedule(filter); err != nil {
 		log.Error(err)
 	}
 
 	os.Exit(1)
 }
 
-func runUpgradesOnSchedule(c *cobra.Command, filter t.Filter) error {
+func runUpgradesOnSchedule(filter t.Filter) error {
```

Comment:
Any specific reason why you prefer to dig config values out from viper rather than passing the command as in `Run` and `PreRun`?

Reply:
the command only has a reference to the flags, but I don't think that was even used before in this, instead it relied on file-global vars. it should probably just be included in the `WatchConfig` struct and have it passed instead